### DISCUSSION TITLE
Blob storage -- use custom domains?

### DISCUSSION THREAD

**Discussion by zeev-code** [vercel/next.js#71421] [Help]
### Summary

Hi,
I didn't find anything on how to use custom domains for linking to documents in Blob storage (I'd like to do my.custom.domain instead of the public.blob.vercel-storage part of the URL).

Is it possible to do this somehow? If not, are there plans for adding this in the future?

Thank you for your help!

Florian

PS.:  v0 suggests  `customDomain=my.custom.domain` but this does not exist in `PutCommandOptions`

**Comment by jeevanpillay** [comment]
same here. would love this feature. 

right now cname redirect seems like the way to go. its more like an alias. not as efficient as a A record though but should suffice.

**Reply by RobertB4** [reply]
Can you elaborate on how you achieved this?
I get an SSL error (ERR_SSL_VERSION_OR_CIPHER_MISMATCH) when I try to add a cname.

**Reply by zeev-code** [reply]
I did not achieve anything, adding custom domains didn't work.
In case by "how I achieved this" you meant what question I asked v0: I simply asked "how can i add custom domains for linking to documents in Blob storage"

**Reply by yihuanlin** [reply]
> same here. would love this feature.
> 
> right now cname redirect seems like the way to go. its more like an alias. not as efficient as a A record though but should suffice.

CNAME did not seem to work for me. It is banned by my DNS provider Cloudflare for Cross-User Request (I guess blob is also hosted on CF). I am using rewrite for now.

> Error 1014
> CNAME Cross-User Banned
> What happened?
> 
> You've requested a page on a website that is part of the [Cloudflare](https://www.cloudflare.com/5xx-error-landing/) network. The host is configured as a CNAME across accounts on Cloudflare, which is not allowed by Cloudflare's security policy.
> What can I do?
> 
> If this is an R2 custom domain, it may still be initializing. If you have attempted to manually point a CNAME DNS record to your R2 bucket, you must do it using a custom domain. Refer to [R2's documentation](https://developers.cloudflare.com/r2/data-access/public-buckets/#connect-your-bucket-to-a-custom-domain) for details.
> 
> [Visit our website](https://www.cloudflare.com/5xx-error-landing/) to learn more about Cloudflare.

**Comment by mdvertola** [comment]
I have not been able to find documentation on custom blob domains on vercel anywhere but here is my workaround:

i created a route in my next app `/api/files/[...]` then i use this to proxy vercel blob object/url and return the file at this endpoint. This way all of my files are served from my domain and i dont need to worry about certs, dns records, etc.

my primary issue was sending emails with photos and having a domain mismatch. This pattern was holding up a lot of my emails and getting me triggered in quarantine/spam/promotions. 

The solution is not perfect and it does add some additional requests to my server but it does also allow for a bit more granular access control systems as well. eg i can use auth js in this api route to lock files down on a per user basis.

**Comment by jpmelnik** [comment]
Hello,

Are there any updates on native support for custom domains in Vercel Blob Storage?

Is this feature planned or on the roadmap?

Thank you!
